draft-ietf-quic-tls.md

```diff
@@ -440,73 +461,32 @@ Important:
   STREAM frame that carries the Finished message in multiple packets.  This
   enables immediate server processing for those packets.
 
+### Encryption Level Changes
 
-### Source Address Validation
-
-During the processing of the TLS ClientHello, TLS requests that the transport
-make a decision about whether to request source address validation from the
-client.
-
-An initial TLS ClientHello that resumes a session includes an address validation
-token in the session ticket; this includes all attempts at 0-RTT.  If the client
-does not attempt session resumption, no token will be present.  While processing
-the initial ClientHello, TLS provides QUIC with any token that is present. In
-response, QUIC provides one of three responses:
-
-* proceed with the connection,
-
-* ask for client address validation, or
-
-* abort the connection.
-
-If QUIC requests source address validation, it also provides a new address
-validation token.  TLS includes that along with any information it requires in
-the cookie extension of a TLS HelloRetryRequest message.  In the other cases,
-the connection either proceeds or terminates with a handshake error.
-
-The client echoes the cookie extension in a second ClientHello.  A ClientHello
-that contains a valid cookie extension will always be in response to a
-HelloRetryRequest.  If address validation was requested by QUIC, then this will
-include an address validation token.  TLS makes a second address validation
-request of QUIC, including the value extracted from the cookie extension.  In
-response to this request, QUIC cannot ask for client address validation, it can
-only abort or permit the connection attempt to proceed.
-
-QUIC can provide a new address validation token for use in session resumption at
-any time after the handshake is complete.  Each time a new token is provided TLS
-generates a NewSessionTicket message, with the token included in the ticket.
-
-See {{client-address-validation}} for more details on client address validation.
-
-
-### Key Ready Events
-
-TLS provides QUIC with signals when 0-RTT and 1-RTT keys are ready for use.
+At each change of encryption level in either direction, TLS signals
+QUIC, providing the new level and the encryption keys. It will also
```

Comment:
This trails off into nothing.  Did some text get lost?

Reply:
Yes. I'll try to figure out what it was

Reply:
Just removed.